js/sketchfab.js

```diff
@@ -119,7 +120,13 @@ SketchfabPlugin.prototype.onSubmit = function onSubmit() {
 	var modelId = url.substring(url.lastIndexOf('/') + 1);
 	var text = this.renderText(modelId);
 
-	this.insertAtCursor(text);
+	if ((this.mode || '').toLowerCase() === 'wysiwyg') {
+		// Undocumented event that's supported by advanced editor only.
+		this.$editor.trigger('appendHtml', text);
+	} else {
+		this.insertAtCursor(text);
```

Comment:
The formulation we usually use is: `$(editor).replaceSelectedText(text);` which should likewise be a bit more robust in what it can handle.